sync/src/test/java/tech/pegasys/artemis/sync/BlockPropagationManagerTest.java

```diff
@@ -155,6 +168,78 @@ public void onBlockImported_withPendingBlocks() throws Exception {
     assertThat(pendingBlocks.size()).isEqualTo(0);
   }
 
+  @Test
+  public void onBlockImportFailure_withPendingDependantBlocks() throws Exception {
+    final int invalidChainDepth = 3;
+    final List<SignedBeaconBlock> invalidBlockDescendants = new ArrayList<>(invalidChainDepth);
+
+    final SignedBeaconBlock invalidBlock =
+        remoteChain.createBlockAtSlotFromInvalidProposer(incrementSlot());
+    Bytes32 parentBlockRoot = invalidBlock.getMessage().hash_tree_root();
+    for (int i = 0; i < invalidChainDepth; i++) {
+      final UnsignedLong nextSlot = incrementSlot();
+      final SignedBeaconBlock block =
+          DataStructureUtil.randomSignedBeaconBlock(nextSlot.longValue(), parentBlockRoot, i);
+      invalidBlockDescendants.add(block);
+      parentBlockRoot = block.getMessage().hash_tree_root();
+    }
+
+    // Gossip all blocks except the first
+    invalidBlockDescendants.stream().map(GossipedBlockEvent::new).forEach(localEventBus::post);
+    assertThat(importedBlocks.get()).isEmpty();
+    assertThat(pendingBlocks.size()).isEqualTo(invalidChainDepth);
+
+    // Gossip next block, causing dependent blocks to be dropped when the import fails
+    localEventBus.post(new GossipedBlockEvent(invalidBlock));
+    assertThat(importedBlocks.get()).isEmpty();
+    assertThat(pendingBlocks.size()).isEqualTo(0);
+
+    // If any invalid block is again gossiped, it should be ignored
+    invalidBlockDescendants.stream().map(GossipedBlockEvent::new).forEach(localEventBus::post);
+    assertThat(importedBlocks.get()).isEmpty();
+    assertThat(pendingBlocks.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void onBlockImportFailure_withUnconnectedPendingDependantBlocks() throws Exception {
+    final int invalidChainDepth = 3;
+    final List<SignedBeaconBlock> invalidBlockDescendants = new ArrayList<>(invalidChainDepth);
+
+    final SignedBeaconBlock invalidBlock =
+        remoteChain.createBlockAtSlotFromInvalidProposer(incrementSlot());
+    Bytes32 parentBlockRoot = invalidBlock.getMessage().hash_tree_root();
+    for (int i = 0; i < invalidChainDepth; i++) {
+      final UnsignedLong nextSlot = incrementSlot();
+      final SignedBeaconBlock block =
+          DataStructureUtil.randomSignedBeaconBlock(nextSlot.longValue(), parentBlockRoot, i);
+      invalidBlockDescendants.add(block);
+      parentBlockRoot = block.getMessage().hash_tree_root();
+    }
+
+    // Gossip all blocks except the first
```

Comment:
I think this should be:
```suggestion
    // Gossip all blocks except the first two
```